### PR TITLE
Allow editing incident updates

### PR DESCRIPTION
**Pull request by nstapelbroek**
HI! :raising_hand_man: 

After reading #2618 I decided to give it a shot.

### What has been done:
- Created controller actions and views for editing an incident update;
- Added a query scope to find all incident updates by an incident;
- Placed a link in the frontend to the edit incident update page;
- Removed the update button from the incident's page;

### How to test:
- Checkout PR
- Make sure [your Cachet is up and running](https://docs.cachethq.io/docs/installing-cachet)
- Make sure you cachet has some incidents or updates, you can use the `php artisan cachet:seed` if required.
- Acknowledge the new incident update editing functionality.

### Notes:
- Clicking on an incident in the dashboard now routes you to it's updates page, I'm not sure if this is then intended flow you want users to take. Let me know if you have any other wishes. 
- In addition, I'm not sure about separating the controller or moving the view templates. If this setup doesn't work for you. Let me know.
- I'm would love to write some acceptance / functional tests to prove the functionality is working. But I could not find them in the project. Any suggestions on how you want this tested?

### CODE REVIEW

**Comment by jbrooksuk** [comment]
This is amazing, thank you so much for doing this!

Some comments:

> Clicking on an incident in the dashboard now routes you to it's updates page, I'm not sure if this is then intended flow you want users to take. Let me know if you have any other wishes.

I didn't read this until I couldn't figure out how to update incidents. We could do a couple of things to make this more obvious:

1. Add a chevron to the end of the Incident name so it's clear there is an action behind it.
2. Add a button next to Edit/Delete as "Manage Updates" or something?

> I'm would love to write some acceptance / functional tests to prove the functionality is working. But I could not find them in the project. Any suggestions on how you want this tested?

Yeah, we don't really do this yet. I'd quite like to get Dusk setup at some point to help us with this, but only once the UI has been refreshed.

I'll leave some code comments now.

P.S. don't worry about fixing StyleCI as it'll automatically push the fixes after merge 👍

File: app/Http/Controllers/Dashboard/IncidentController.php [hunk 1]
```diff
@@ -293,9 +295,23 @@ public function editTemplateAction(IncidentTemplate $template)
      *
      * @return \Illuminate\View\View
      */
-    public function showIncidentUpdateAction(Incident $incident)
+    public function showIncidentUpdates(Incident $incident)
     {
-        return View::make('dashboard.incidents.update')->withIncident($incident);
+        $updates = IncidentUpdate::byIncident($incident)->orderBy('created_at', 'desc')->get();
```
Comment (jbrooksuk) [inline app/Http/Controllers/Dashboard/IncidentController.php]: I don't think you need to do this as within the view you can do:

```blade
$incident->updates...
```

File: resources/views/single-incident.blade.php [hunk 2]
```diff
@@ -31,6 +31,11 @@
                 <div class="col-xs-10 col-xs-offset-2 col-sm-11 col-sm-offset-0">
                     <div class="panel panel-message incident">
                         <div class="panel-body">
+                            @if($current_user)
+                                <div class="pull-right btn-group">
```
Comment (jbrooksuk) [inline resources/views/single-incident.blade.php]: We don't need to indent under the condition.

**Comment by nstapelbroek** [comment]
Thanks for the feedback :muscle: !

After looking to some other pages I've decided to remove the anchors from any item in the incident/incident updates overview. Because pages like the Metrics, Maintenance and Components share the same behavior I thought consistency could be a good argument to remove the anchors.

If you do not agree: That's fair. Let me know how you'd like it and I'll make it happen.

**Comment by jbrooksuk** [comment]
This is awesome, thanks!

I'm really excited to see what else you contribute :)

**Pull request merged**